src/reducers/UserReducer.ts

```diff
@@ -22,16 +22,17 @@ const userReducer = (state: DefaultStateI = defaultState, action: UserDispatchTy
     switch (action.type) {
         case USER_FAIL:
             return {
-                loading: false,
+                ...defaultState,
```

Comment:
`defaultState` refers to initial state,
`state` refers to current state. 
use current state instead of initial state.

Rest of code looks good.